File: src/services/userService.ts
import api from './api';
import type { User, PaginatedResponse, PaginationParams } from '@/types';

export const userService = {
  async getAll(params?: Partial<PaginationParams>): Promise<PaginatedResponse<User>> {
    const { data } = await api.get('/users', { params });
    return data;
  },

  async getTechnicians(): Promise<User[]> {
    const { data } = await api.get('/users', { params: { role: 'technician' } });
    return data.data || data;
  },

  async getById(id: string): Promise<User> {
    const { data } = await api.get(`/users/${id}`);
    return data;
  },

  async create(user: Partial<User> & { password?: string }): Promise<User> {
    const { data } = await api.post('/users', user);
    return data;
  },

  async update(id: string, user: Partial<User>): Promise<User> {
    const { data } = await api.put(`/users/${id}`, user);
    return data;
  },

  async delete(id: string): Promise<void> {
    await api.delete(`/users/${id}`);
  },
};

File: FieldServiceManangement-main/src/main/java/com/fieldservicemanagement/field_service_management/service/UserService.java
package com.fieldservicemanagement.field_service_management.service;

import com.fieldservicemanagement.field_service_management.entity.Users;
import com.fieldservicemanagement.field_service_management.repository.UserRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.NoSuchElementException;
import java.util.Optional;

@Service
public class UserService {

    private final UserRepository userRepository;

    public UserService(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    public List<Users> getAll() {
        return userRepository.findAll();
    }

    public List<Users> getByRole(String role) {
        return userRepository.findAll().stream()
                .filter(u -> u.getRole().equalsIgnoreCase(role))
                .toList();
    }

    public Users getById(Long id) {
        return userRepository.findById(id)
                .orElseThrow(() -> new NoSuchElementException("User not found with id: " + id));
    }

    public Optional<Users> getByEmail(String email) {
        return userRepository.findByEmail(email);
    }

    public Users create(Users user) {
        return userRepository.save(user);
    }

    public Users update(Long id, Users userDetails) {
        Users user = getById(id);
        user.setFirstName(userDetails.getFirstName());
        user.setLastName(userDetails.getLastName());
        user.setEmail(userDetails.getEmail());
        user.setRole(userDetails.getRole());
        user.setPasswordHash(userDetails.getPasswordHash());
        return userRepository.save(user);
    }

    public void delete(Long id) {
        userRepository.deleteById(id);
    }
}